src/Template/User/profile.ctp

```diff
@@ -302,106 +306,180 @@ $this->set('title_for_layout', h($this->Pages->formatTitle($title)));
         ?>
     </div>
 
-    <div ng-cloak class="section with-title-button md-whiteframe-1dp">
-        <div layout="row" layout-align="start center">
-            <h2 flex><?= __('Languages'); ?></h2>
-            <?php
-            if ($username == $currentMember) {
-                $addLangUrl = $this->Url->build(array(
-                    'controller' => 'user',
-                    'action' => 'language'
-                ));
-                ?>
-                <div layout="row" layout-align="end center">
-                    <md-button aria-label="<?= __('Add a language') ?>" class="md-primary md-raised" href="<?= $addLangUrl ?>">
-                        <?= __('Add a language') ?>
-                    </md-button>
-                </div>
-            <?php } ?>
-        </div>
-
-        <?php
-        if (empty($userLanguages))
-        {
-            echo '<p>';
-            __('No language added.');
-            echo '</p>';
+<?php
+$userLanguages = htmlspecialchars(json_encode($userLanguages), ENT_QUOTES, 'UTF-8');
+?>
+    <div class="section md-whiteframe-1dp"
+         ng-cloak
+         ng-controller="LanguageController as vm"
+         ng-init="vm.init(<?= $userLanguages ?>)">
+        <h2><?= __('Languages'); ?></h2>
 
-            echo '<p>';
+        <p ng-if="vm.langs.length == 0">
```

Comment:
In the Javascript world, the recommendation is to use `===` by default, rather than `==`. At least it seems that developers have had enough headaches from using `==` that there's plenty of articles explaining why `===` is almost always the correct way to test equality.

In this particular case here I don't think there will be unwanted side effects, but it's just to make sure you're aware, since I've seen you have been using double equality elsewhere.